emulator.c

```diff
@@ -81,6 +87,9 @@ execute_instruction(i8080 *cpu, uint8_t opcode)
       }
     case 0x11: // NOLINT
       printf("LXI D");
+      cpu->d = cpu_read_mem(cpu, cpu->pc + 1);
+      cpu->e = cpu_read_mem(cpu, cpu->pc + 2);
```

Comment:
I _think_  your registers are reversed here and you need  cpu->e = cpu_read_mem(cpu, cpu->pc + 1) etc due to little endianness.

Reply:
Yeah e is pc + 1 and d is pc + 2.